velox/functions/prestosql/aggregates/VarianceAggregates.cpp

```diff
@@ -230,14 +230,12 @@ class VarianceAggregate : public exec::Aggregate {
       });
     } else if (!exec::Aggregate::numNulls_ && decodedRaw_.isIdentityMapping()) {
       const T* data = decodedRaw_.data<T>();
-      VarianceAccumulator accData;
-      rows.applyToSelected([&](vector_size_t i) { accData.update(data[i]); });
```

Comment:
I believe this is a performance optimization. It is cheaper to aggregate all values "locally" first and update accumulator once.

Reply:
Thanks for pointing this out. Do you think it’s necessary to proceed with this change, such as creating relevant benchmarks to evaluate its performance impact?

Reply:
If we want to proceed, it would be helpful to understand perf impact.

Reply:
@zhli1142015 Thank you for measuring perf impact. Would you clarify what do we gain at the expense of 10% perf regression?

Reply:
This change aligns the computation behavior more closely with Spark. It only affects very small decimal values. So far, it primarily helps our unit tests pass and is unlikely to have any real-world impact.

Reply:
Yes, will update the test case in our side.

Reply:
@mbasmanova , thank you for reviewing and pointing out the performance impact of this change.